hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/snapshot/SnapshotManager.java

```diff
@@ -267,6 +280,20 @@ public INodeDirectory getSnapshottableRoot(final INodesInPath iip)
     return dir;
   }
 
+  public void assertMarkedAsDeleted(INodesInPath iip, String snapshotName)
+      throws IOException {
+    final INodeDirectory dir = getSnapshottableRoot(iip);
+    final Snapshot.Root snapshotRoot = dir.getDirectorySnapshottableFeature()
+        .getSnapshotByName(dir, snapshotName)
+        .getRoot();
+
+    if (snapshotRoot.isMarkedAsDeleted()) {
```

Comment:
The check here should be !snapshotRoot.isMarkedAsDeleted() ?

Reply:
Good catch. Thanks.